src/ansys/dpf/core/custom_type_fields_container.py

```diff
@@ -0,0 +1,589 @@
+        global server.
+
+    """
+
+    def __init__(
+        self,
+        unitary_type=None,
+        custom_type_fields_container=None,
+        server=None,
+    ):
+        super().__init__(collection=custom_type_fields_container, server=server)
+        if self._internal_obj is None:
+            if self._server.has_client():
+                self._internal_obj = (
+                    self._api.collection_of_custom_type_field_new_on_client(  # TODO
+                        self._server.client
+                    )
+                )
+            else:
+                self._internal_obj = self._api.collection_of_custom_type_field_new()
+        self._type = unitary_type
+        self._component_index = None  # component index
+        self._component_info = None  # for norm/max/min
+
+    def create_subtype(self, obj_by_copy) -> CustomTypeField:
+        return CustomTypeField(unitary_type=self._type, field=obj_by_copy, server=self._server)
+
+    def get_fields_by_time_complex_ids(
+        self, timeid: Union[int, None] = None, complexid: Union[int, None] = None
+    ) -> list[CustomTypeField]:
+        """Retrieve custom type fields at a requested time ID or complex ID.
+
+        Parameters
+        ----------
+        timeid:
+            Time ID or frequency ID, which is the one-based index of the
+            result set.
+        complexid:
+            Complex ID, where ``1`` is for imaginary and ``0`` is for real.
+
+        Returns
+        -------
+        fields :
+            Custom type fields corresponding to the request.
+
+        """
+        label_space = self.__time_complex_label_space__(timeid, complexid)
+        return super()._get_entries(label_space)
+
+    def get_field_by_time_complex_ids(
+        self, timeid: Union[int, None] = None, complexid: Union[int, None] = None
+    ) -> CustomTypeField:
+        """Retrieve a custom type field at a requested time ID or complex ID.
+
+        An exception is raised if the number of fields matching the request is
+        greater than one.
+
+        Parameters
+        ----------
+        timeid:
+            Time ID or frequency ID, which is the one-based index of the
+            result set.
+        complexid:
+            Complex ID, where ``1`` is for imaginary and ``0`` is for real.
+
+        Returns
+        -------
+        fields:
+            Custom type field corresponding to the request
+
+        """
+        label_space = self.__time_complex_label_space__(timeid, complexid)
+        return super()._get_entry(label_space)
+
+    def __time_complex_label_space__(self, timeid: int = None, complexid: int = None) -> dict:
+        label_space = {}
+        if timeid is not None:
+            label_space["time"] = timeid
+        if complexid is not None:
+            label_space["complex"] = complexid
+        return label_space
+
+    def get_fields(self, label_space: dict[str, int]) -> list[CustomTypeField]:
```

Comment:
I guess that it is intentional, but wouldn't it make more sense to specialize the apis? I mean, `get_customfields` instead of `get_fields`, and the same for the rest

Reply:
@rafacanton yes, it was intentional on my part to have its usage as similar as possible to that of a `FieldsContainer` as to me they are two different implementations of the same thing, so they should have as much a similar API as possible.
I had to remove those methods though as server-side none of the `collection_capi` methods understand a CustomTypeFieldsContainer.